code/modules/shuttle/computer.dm

```diff
@@ -29,17 +31,20 @@
 	var/list/options = params2list(possible_destinations)
 	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
 	data["docked_location"] = M ? M.get_status_text_tgui() : "Unknown"
-	data["status"] = M.mode == SHUTTLE_IGNITING ? "Igniting" : M.mode != SHUTTLE_IDLE ? "In Transit" : "Idle"
 	data["locations"] = list()
-	data["locked"] = FALSE
+	data["locked"] = locked
 	data["authorization_required"] = admin_controlled
 	data["timer_str"] = M ? M.getTimerStr() : "00:00"
 	data["destination"] = destination
-	if(admin_controlled)
-		data["status"] = "Unauthorized Access"
 	if(!M)
 		data["status"] = "Missing"
 		return data
+	if(admin_controlled)
+		data["status"] = "Unauthorized Access"
+	else if(locked)
+		data["status"] = "Locked"
+	else
+		data["status"] = M.mode == SHUTTLE_IGNITING ? "Igniting" : M.mode != SHUTTLE_IDLE ? "In Transit" : "Idle"
```

Comment:
This would be cleaner as an if or a switch, rather than two ternaries.

Reply:
Used a switch instead.